packages/cli/src/commands/keys/add.ts

```diff
@@ -0,0 +1,111 @@
+import {Args, ux} from '@oclif/core'
+import color from '@heroku-cli/color'
+import {flags, Command} from '@heroku-cli/command'
+import * as  inquirer from 'inquirer'
+import * as  path from 'path'
+import * as os from 'os'
+import * as fs from 'fs-extra'
+
+function sshKeygen(file: string, quiet: boolean) {
+  const spawn = require('child_process').spawn
+  return new Promise(function (resolve, reject) {
+    spawn('ssh-keygen', ['-o', '-t', 'rsa', '-N', '', '-f', file], {stdio: quiet ? null : 'inherit'})
+      .on('close', (code: number) => code === 0 ? resolve(null) : reject(code))
+  })
+}
+
+async function confirmPrompt(message: string) {
+  if (process.stdin.isTTY) {
+    return inquirer.prompt([{
+      type: 'confirm',
+      name: 'yes',
+      message: message,
+    }])
+  }
+
+  const data = await ux.prompt(message + ' [Y/n]')
+  return {yes: /^y(es)?/i.test(data)}
+}
+
+export default class Add extends Command {
+  static description = 'add an SSH key for a user'
+  static help = 'if no KEY is specified, will try to find ~/.ssh/id_rsa.pub'
+  static example = `$ heroku keys:add
+Could not find an existing public key.
+Would you like to generate one? [Yn] y
+Generating new SSH public key.
+Uploading SSH public key /.ssh/id_rsa.pub... done
+
+$ heroku keys:add /my/key.pub
+Uploading SSH public key /my/key.pub... done`
+
+  static flags = {
+    quiet: flags.boolean({hidden: true}),
+    yes: flags.boolean({char: 'y', description: 'automatically answer yes for all prompts'}),
+  }
+
+  static args = {
+    key: Args.string(),
+  }
+
+  async run() {
+    const {flags, args} = await this.parse(Add)
+    const sshdir = path.join(os.homedir(), '.ssh')
+
+    const generate = async function () {
+      await fs.ensureDir(sshdir, {mode: 0o700})
+      await sshKeygen(path.join(sshdir, 'id_rsa'), flags.quiet)
+    }
+
+    const findKey = async function () {
+      const defaultKey = path.join(sshdir, 'id_rsa.pub')
+      if (!(await fs.pathExists(defaultKey))) {
+        console.error('Could not find an existing SSH key at ' + path.join('~', '.ssh', 'id_rsa.pub'))
```

Comment:
Not sure about swapping `cli.console.error` for `console.error` but there are no `ux` equivalents that write to stderr without adding a prefix "Error: "

Reply:
Is there a reason besides not wanting to change the messaging for not adding the "Error: " prefix?

Reply:
It has been getting written to stderr, which is the standard for passing messages. My thinking is that changing it could have unintended consequences. I'm not sure about this, and maybe it should be an intended consequence since this will be released along with a major version.